docs/resources/samples/excel-samples.md

```diff
@@ -146,6 +146,68 @@ function main(workbook: ExcelScript.Workbook) {
 }
 ```
 
+### Unhide All Rows and Columns
+
+This script get the worksheet's used range, checks if there are any hidden rows and columns, and unhides them. 
+
+```Typescript
+function main(workbook: ExcelScript.Workbook) {
+    // Get the currently selected sheet.
+    const selectedSheet = workbook.getActiveWorksheet();
+
+    // Get the entire data range.
+    const range = selectedSheet.getUsedRange();
+
+    // If the used range is empty, end the script.
+    if (!range) {
+      console.log(`No data on this sheet.`)
+      return;
+    }
+
+    // Log the address of the used range.
+    console.log(`Used range for the worksheet: ${range.getAddress()}`);
+
+    // If no columns are hidden, log message, else, unhide columns
+    if (range.getColumnHidden() == false) {
+      console.log(`No columns hidden`);
+    } else {
+      range.setColumnHidden(false);
+    }
+
+    // If no rows are hidden, log message, else, unhide rows.
+    if (range.getRowHidden() == false) {
+      console.log(`No rows hidden`);
+    } else {
+      range.setRowHidden(false);
+    }
+```
+
+### Freeze Currently Selected Cells
+
+This script checks what cells are currently selected, and freezes that selection so those cells are always visible.
+
+```Typescript
+function main(workbook: ExcelScript.Workbook) {
+    // Get the currently selected sheet.
+    const selectedSheet = workbook.getActiveWorksheet();
+
+    // Get the current selected range.
+    const selectedRange = workbook.getSelectedRange();
+
+    // If no cells are selected, end the script. 
+    if (selectedRange == null) {
+      console.log(`No cells in worksheet selected.`);
+      return;
+    }
```

Comment:
getSelectedRange() returns a ExcelScript.Range object rather than a boolean. The if statement change would cause an error along the lines of a null pointer exception if it tried to "not" an null object. Link to docs: https://docs.microsoft.com/en-us/javascript/api/office-scripts/excelscript/excelscript.workbook?view=office-scripts#excelscript-excelscript-workbook-getselectedrange-member(1)

Reply:
This edit shouldn't change what error gets thrown. Either way, the `get` call is made earlier and this value is compared.  This just lets the `if` statement work regardless of whether `null` or `undefined` is returned.